Damping.pyx

```diff
@@ -36,22 +49,470 @@ cdef class Damping:
+                            for k in xrange(kmin, kmax):
+                                ijk = ishift + jshift + k
+                                PV.tendencies[var_shift + ijk] -= (PV.values[var_shift + ijk] - 0.0) * self.gamma_zhalf[k] 
+
+            elif var_name == 'u' or var_name == 'v':
+                with nogil:
+                    for i in xrange(imin, imax):
+                        ishift = i * istride
+                        for j in xrange(jmin, jmax):
+                            jshift = j * jstride
+                            for k in xrange(kmin, kmax):
+                                ijk = ishift + jshift + k
+                                PV.tendencies[var_shift + ijk] -= (PV.values[var_shift + ijk] - domain_mean[k]) * self.gamma_z[k]
+            else:
+                with nogil:
+                    for i in xrange(imin, imax):
+                        ishift = i * istride
+                        for j in xrange(jmin, jmax):
+                            jshift = j * jstride
+                            for k in xrange(kmin, kmax):
+                                ijk = ishift + jshift + k
+                                #PV.tendencies[var_shift + ijk] *= self.tend_flat[k]
+                                PV.tendencies[var_shift + ijk] -= (PV.values[var_shift + ijk] - domain_mean[k]) * self.gamma_z[k]
+
+        if 's' in PV.name_index:
+            s_shift = PV.get_varshift(Gr, 's')
+        else:
+            s_shift = PV.get_varshift(Gr, 'thli')
+
+        with nogil:
+            for i in xrange(gw,imax):
+                ishift = i * istride
+                for j in xrange(gw,jmax):
+                    jshift = j * jstride
+                    for k in xrange(gw,kmax):
+
+                        weight = self.tend_flat_half[k]
+
+                        ijk = ishift + jshift + k
+                        p0 = RS.p0_half[k]
+                        rho0 = RS.rho0_half[k]
+                        qt = PV.values[qt_shift + ijk]
+                        qv = qt - DV.values[ql_shift + ijk]
+                        pd = pd_c(p0,qt,qv)
+                        pv = pv_c(p0,qt,qv)
+                        t  = DV.values[t_shift + ijk]
+                        PV.tendencies[s_shift + ijk] =   (weight) * PV.tendencies[s_shift + ijk]
+                        #PV.tendencies[s_shift + ijk] += (sv_c(pv,t) - sd_c(pd,t)) * (self.dt_qg_total[k]* (1.0 -weight) ) + (1.0 - weight) * (cpm_c(qt) * (self.dt_tg_total[k]))/t
+                        #PV.tendencies[qt_shift + ijk] = self.dt_qg_total[k]* (1.0 - weight) + PV.tendencies[qt_shift + ijk] *(weight)
+                        PV.tendencies[qt_shift + ijk] = PV.tendencies[qt_shift + ijk] * (weight)
+                        PV.tendencies[u_shift + ijk] = (weight) * PV.tendencies[u_shift + ijk]
+                        PV.tendencies[v_shift + ijk] = (weight) * PV.tendencies[v_shift + ijk] 
+
+        return
+
+
+
+cdef class RayleighGCMVarying:
```

Comment:
Let's remove `class RayleighGCMVarying`. (It's not used, and if we need it one day, it's probably better to rewrite it.)